lxd/db/query/objects.go

```diff
@@ -31,6 +31,31 @@ func SelectObjects(stmt *sql.Stmt, dest Dest, args ...any) error {
 	return nil
 }
 
+// QueryObjects executes a query from a SQL string which must yield rows with
+// a specific columns schema. It invokes the given Dest hook for each yielded row.
+func QueryObjects(tx *sql.Tx, stmt string, dest Dest, args ...any) error {
```

Comment:
Possibly a point for another day, but how do you find the usability of `SelectObjects` and this new function `QueryObjects`? 

I've always found it a little odd that one has to define a callback function like:

```go
	args := []any{}

	// Dest function for scanning a row.
	dest := func(i int) []any {
		objects = append(objects, Device{})
		return []any{
			&objects[i].ID,
			&objects[i].ReferenceID,
			&objects[i].Name,
			&objects[i].Type,
		}
	}
``` 

Whose return value is then passed to the `rows.Scan()` function.

That is why I added `tx.QueryScan()` a while back, which is similar, but instead defines a callback function that is passed the `rows.Scan()` function and allows it to call it however it wants. The callback function is also defined as returning an `error` so if needed the query result scanning can be ended prematurely if something goes wrong.

 It also avoids the need to define a slice containing the field map for every row, and instead it leaves it to the caller to call `rows.Scan` in which ever way it wants.

What would you say about switching to using `QueryScan` (or something similar) for this function, as well as `SelectObjects`?